### ISSUE TITLE
Semaphore crashes when I try to launch task with non existent ID via API

### ISSUE THREAD

**Issue by strangeman** [ansible-semaphore/semaphore#549]
I launch task with this curl script
```
curl -X POST -v --header 'Content-Type: application/json' \
--header 'Accept: application/json' \
-d '{"template_id": 999, "debug": false, "dry_run": false, "playbook": "", "environment": ""}' \
-H 'Authorization: Bearer XXXXXXXXXXXXXXXXXXXXXXX' \
'http://localhost:8010/api/project/1/tasks'
```

and get following error:

```
2018/06/09 17:26:25 http: panic serving 127.0.0.1:60506: Error 1452: Cannot add or update a child row: a foreign key constraint fails (`semaphore`.`task`, CONSTRAINT `task_ibfk_1` FOREIGN KEY (`template_id`) REFERENCES `project__template` (`id`) ON DELETE CASCADE)
goroutine 224 [running]:
net/http.(*conn).serve.func1(0xc420248640)
	/usr/local/go/src/net/http/server.go:1726 +0xd0
panic(0x9396a0, 0xc4200ae6e0)
	/usr/local/go/src/runtime/panic.go:505 +0x229
github.com/ansible-semaphore/semaphore/api/tasks.AddTask(0xded420, 0xc42047f1c0, 0xc4208c2b00)
	/mnt/src/sources/go/src/github.com/ansible-semaphore/semaphore/api/tasks/http.go:31 +0x612
github.com/ansible-semaphore/semaphore/vendor/github.com/castawaylabs/mulekick.Router.Handle.func1(0xdee660, 0xc420870460, 0xc4208c2b00)
	/mnt/src/sources/go/src/github.com/ansible-semaphore/semaphore/vendor/github.com/castawaylabs/mulekick/router.go:69 +0x138
net/http.HandlerFunc.ServeHTTP(0xc4201ae770, 0xdee660, 0xc420870460, 0xc4208c2b00)
	/usr/local/go/src/net/http/server.go:1947 +0x44
github.com/ansible-semaphore/semaphore/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc42012a000, 0xdee660, 0xc420870460, 0xc4208c2b00)
	/mnt/src/sources/go/src/github.com/ansible-semaphore/semaphore/vendor/github.com/gorilla/mux/mux.go:159 +0xed
github.com/ansible-semaphore/semaphore/vendor/github.com/gorilla/handlers.ProxyHeaders.func1(0xdee660, 0xc420870460, 0xc4208c2900)
	/mnt/src/sources/go/src/github.com/ansible-semaphore/semaphore/vendor/github.com/gorilla/handlers/proxy_headers.go:59 +0x117
net/http.HandlerFunc.ServeHTTP(0xc4201a8b60, 0xdee660, 0xc420870460, 0xc4208c2900)
	/usr/local/go/src/net/http/server.go:1947 +0x44
net/http.(*ServeMux).ServeHTTP(0x10cddc0, 0xdee660, 0xc420870460, 0xc4208c2900)
	/usr/local/go/src/net/http/server.go:2337 +0x130
net/http.serverHandler.ServeHTTP(0xc420130000, 0xdee660, 0xc420870460, 0xc4208c2900)
	/usr/local/go/src/net/http/server.go:2694 +0xbc
net/http.(*conn).serve(0xc420248640, 0xdeeb60, 0xc42047f0c0)
	/usr/local/go/src/net/http/server.go:1830 +0x651
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2795 +0x27b
```

**Comment by twhiston** [comment]
looks like this is caused by `tasks/http.go` l77
```
	if _, err := db.Mysql.Select(&tasks, query, args...); err != nil {
		panic(err)
	}
```
and is due to the general problem that we need to make semaphore less panicky. I guess we need to check for the error type, see if we can filter out real errors from no results and if a real error call `util.LogErrorWithFields`. In both cases the user should get some sensible (ie does not allow attackers to probe for failure conditions) response provided by the api.

**Comment by strangeman** [comment]
Fixed by #551
